src/agents/camp_agent.py

```diff
@@ -1,81 +1,31 @@
-from typing import List
-
 from steamship import Block
-from steamship.agents.functional import FunctionsBasedAgent
-from steamship.agents.schema import AgentContext
-from steamship.agents.schema.message_selectors import MessageWindowMessageSelector
+from steamship.agents.schema import Action, Agent, AgentContext, FinishAction
 
-from schema.game_state import GameState
 from tools.start_quest_tool import StartQuestTool
-from utils.context_utils import get_game_state
-from utils.moderation_utils import is_block_excluded
-
-
-class NonExcludedMessageWindowSelector(MessageWindowMessageSelector):
-    def get_messages(self, messages: List[Block]) -> List[Block]:
-        # choice here: if the user has been submitting a bunch of problematic conversation
-        #              then we are ok "forgetting" the distant context. so, we can trim from
-        #              the discovered window, rather than first filtering and then selecting
-        #              via the window.
-        context_messages = super().get_messages(messages)
-        return [msg for msg in context_messages if not is_block_excluded(msg)]
-
-
-class CampAgent(FunctionsBasedAgent):
-    """The Camp Agent is in charge of your experience while at camp.
 
-    A player is at camp while they are not on a quest.
 
-    CONSIDER: This feels like a FunctionCallingAgent
+class CampAgent(Agent):
+    """The Camp Agent is currently just
+    the default agent for CLI debugging.  It is not used in the web game.
     """
 
-    PROMPT = ""
-
     def __init__(self, **kwargs):
         super().__init__(
             tools=[StartQuestTool()],  # , StartConversationTool()],
-            message_selector=NonExcludedMessageWindowSelector(k=10),
             **kwargs,
         )
 
-    def next_action(self, context: AgentContext):
-        """Choose the next action.
-
-        We defer to the superclass but have to dynamically set our prompt with the context."""
-        game_state = get_game_state(context)
-        self.set_prompt(game_state)
-        return super().next_action(context)
+    def next_action(self, context: AgentContext) -> Action:
+        """Only one action possible - start quest (for CLI usage).  Else return dummy string."""
 
-    def set_prompt(self, game_state: GameState):
-        npcs = []
-        if game_state and game_state.camp and game_state.camp.npcs:
-            npcs = game_state.camp.npcs
+        last_message = ""
+        if last_message_block := context.chat_history.last_user_message:
+            if last_message_block.text is not None:
+                last_message = last_message_block.text
 
-        if npcs:
-            camp_crew = "Your camp has the following people:\n\n" + "\n".join(
-                [f"- {npc.name}" for npc in npcs or []]
-            )
+        if last_message.lower() == "quest":
```

Comment:
question: do we want something like `if "quest" in last_message.lower():` just to provide additional robustness ?

Reply:
Sounds good to me